src/streamlink/plugins/pluzz.py

```diff
@@ -0,0 +1,122 @@
+        'videos': validate.all(
+            [{
+                'format': validate.text,
+                'url': validate.any(
+                    None,
+                    validate.url(),
+                ),
+                'statut': validate.text,
+                'geoblocage': validate.any(
+                    None,
+                    [validate.all(validate.text)]
+                ),
+                'plages_ouverture': validate.all(
+                    [{
+                        'debut': validate.any(
+                            None,
+                            int
+                        ),
+                        'fin': validate.any(
+                            None,
+                            int
+                        )
+                    }]
+                )
+            }]
+        )
+    })
+
+
+    @classmethod
+    def can_handle_url(cls, url):
+        return Pluzz._url_re.match(url)
+
+
+    def _get_streams(self):
+        # Retrieve geolocation data
+        res = http.get(self.GEO_URL)
+        geo = http.json(res, schema=self._geo_schema)
+        country_code = geo['reponse']['geo_info']['country_code']
+
+        # Retrieve URL page and search for video ID
+        res = http.get(self.url)
+        match = self._video_id_re.search(res.text)
+        if match is None:
+            return
+        video_id = match.group('video_id')
+
+        # Retrieve SWF player URL
+        match = self._player_re.search(res.text)
+        swf_url = None
+        if match is not None:
+            player_url = 'http:' + match.group('player')
+            res = http.get(player_url)
+            match = self._swf_re.search(res.text)
+            if match is not None:
+                swf_url = os.path.dirname(player_url) + match.group('swf')
+
+        res = http.get(self.API_URL.format(video_id))
+        videos = http.json(res, schema=self._api_schema)
+        now = time.time()
+
+        for video in videos['videos']:
+            # Check whether video format is available
+            if video['statut'] != 'ONLINE':
+                continue
+
+            # Check whether video format is geo-locked
+            if video['geoblocage'] is not None and country_code not in video['geoblocage']:
+                continue
+
+            # Check whether video format is expired
+            available = False
+            for interval in video['plages_ouverture']:
+                available = interval['debut'] <= now <= interval['fin']
+                if available:
+                    break
+            if not available:
+                continue
+
+            video_url = video['url']
+            # TODO: add DASH streams once supported
+            if '.f4m' in video_url and swf_url is not None:
+                res = http.get(self.HDS_TOKEN_URL.format(video_url))
+                video_url = res.text
+                for stream in HDSStream.parse_manifest(self.session, video_url, pvswf=swf_url).items():
```

Comment:
Have you tried using `AkamaiHDStream` for troublesome streams? They might have extra auth/verification.